### ISSUE TITLE
consider adding a "consistent file naming" check

### ISSUE THREAD

**Issue by jdanyow** [MeirionHughes/aurelia-template-lint#79]
Rule would be something like:

>*for all `.js` or `.ts` files with a corresponding `.html` file, verify the filename matches the name of the first export*:

consider the following file:

```js
export FooBar {
}
```

The expected filename would be `foo-bar.js` or `FooBar.js` depending on the naming convention people are using.

**Comment by atsu85** [comment]
Simple naming rule for all TypeScript files can be easily implemented with tslint rules, but I'd like a bit more complex naming conventions:
1) prefix-xxx-yyy - custom elements should have file names with the same name as custom element tag names(for intuitive navigation between files in ide), and class names should be named similarly, except with UpperCamelCase: PrefixXxxYyy or PrefixXxxYyyCustomElement.
2) xxxYyy (or xxxYyyViewModel) - viewmodels (not used as custom elements, but for routing) should have file names with lowerCamelCase (to distinguish them from custom elements and regular classes not used by Aurelia) and class names with UpperCamelCase.
3) XxxYyy - files containing regular TypeScript classes should be named with UpperCamelCase just like class file they export.

This convention has served my team very well:
1) sort but easily distinguishable file names (grouping files by feature, not by layer, so it scales well for large projects and keeps related code close to each other)
2) easy to navigate between files
3) class names correspond to file names (with case convertion differences)

**Comment by MeirionHughes** [comment]
@jdanyow could you have a quick look at https://github.com/MeirionHughes/aurelia-template-lint/issues/113 please? I thought the convention was (As you say) first exported class should be the VM; but apparently that convention isn't used for routers, oddly. Is that (router not using first export) a bug?
